gaphor/diagram/general/generalpropertypages.py

```diff
@@ -88,3 +97,76 @@ def construct(self):
     def _on_field_change(self, entry, field_name):
         text = entry.get_text()
         setattr(self.item, field_name, text)
+
+
+FILTER_IMAGES = [
+    ("Images", "*.png, *.jpg, *.jpeg", "image/*"),
+]
+
+
+@PropertyPages.register(PictureItem)
+class PicturePropertyPage(PropertyPageBase):
+    """Edit picture settings"""
+
+    def __init__(self, subject):
+        self.subject = subject
+        self.watcher = subject and subject.watcher()
+
+    def construct(self):
+        subject = self.subject
+
+        if not subject:
+            return
+
+        builder = new_builder(
+            "picture-editor",
+            signals={
+                "select-picture": (self._on_select_picture_clicked,),
+                "set-default-size": (self._on_default_size_clicked),
+            },
+        )
+        return builder.get_object("picture-editor")
+
+    @transactional
+    def _on_select_picture_clicked(self, button):
+        open_file_dialog(
+            gettext("Select a picture..."), self.open_files, filters=FILTER_IMAGES
```

Comment:
Since we aren't passing in the parent window, we are getting `Gtk-Message: 12:26:09.933: GtkDialog mapped without a transient parent. This is discouraged.`

I am not sure the best way to get the parent window of the propertypage to be honest, maybe @amolenaar has an idea.

Reply:
The file filter also isn't working on Windows (it seems to work in Linux). I think we may need to modify it so that the filters are passed as a list instead of a comma separated string.

Reply:
Tried the replace it with list but received a python error. Now trying it out with a simple space separated string, maybe it works.

Reply:
I am not a big fan of singletons, but it would be nice to be able to acquire a reference to the `Application` and then simply ask for the `main_window`.

Reply:
@sz332 I pushed a fix for the filter not working on Windows by using the built-in [add_pixbuf_formats](https://docs.gtk.org/gtk4/method.FileFilter.add_pixbuf_formats.html) filter. Please let me know what you think.

@amolenaar, any ideas on how we can get the parent window here? I don't feel very comfortable with us merging this with the current GtkDialog warning.

Reply:
I think it looks totally fine, especially that it is being supported "natively".